src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1555,5 +1556,26 @@ private void UpgradeButton_Click(object sender, RoutedEventArgs e)
             })
             .PostOnFailure(nameof(PackageManagerControl), nameof(UpgradeButton_Click));
         }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (disposing)
+            {
+                CleanUp();
```

Comment:
`CleanUp()` starts async work in the background. If this object gets disposed, then all the fields that `CleanUp` references may be null and cause a bunch of null reference exceptions.

Reply:
`CleanUp()` starts async work to invoke `IVsWindowSearchHost.TerminateSearch` method in `JTF.Run` context which blocks the caller until `async` operation ran to completion.  Hence, I think `null` reference exception won't be thrown. Please let me know if I am missing something.
https://github.com/NuGet/NuGet.Client/blob/3b1d6b8ab827236ea92fb25fb4e6c29596cf6c47/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L1344-L1354

Reply:
You're right, I was wrong. `jtf.Run` will block until the async method is complete.